screens/Contacts/ContactGraph/ContactGraph.tsx

```diff
@@ -1,88 +1,131 @@
-import { View, TextInput, TouchableWithoutFeedback, Keyboard } from "react-native"
+import {
+  View,
+  TextInput,
+  TouchableWithoutFeedback,
+  Keyboard,
+} from "react-native"
 import { styles } from "./styles"
-import Graph, { Node } from "../../../components/Graph/Graph"
+import Graph, {
+  getNodeById,
+  getNodes,
+  Node,
+} from "../../../components/Graph/Graph"
 
 import ForceDirectedGraph from "../../../components/Graph/ForceDirectedGraph/ForceDirectedGraph"
-import { useState } from "react"
+import React, { useState } from "react"
 
 import NodeModal from "../../../components/Graph/NodeModal/NodeModal"
 
-import { graph, constrainedNodeId } from "./mockGraph"
-interface ContactGraphProps{
-  onContactPress: (uid : string) => void
+interface ContactGraphProps {
+  onContactPress: (uid: string) => void
+  graph: Graph
+  userId: string
 }
 
-const ContactGraph = ({onContactPress} : ContactGraphProps) => {
-
+/**
+ * ContactGraph component
+ * @param onContactPress - Function to call when the contact is pressed
+ * @param graph - The graph to display
+ * @param userId - The unique identifier of the user
+ * @returns The ContactGraph component
+ */
+const ContactGraph = ({ onContactPress, graph, userId }: ContactGraphProps) => {
+  // State to store the text of the search bar
   const [searchText, setSearchText] = useState("")
 
-  const [clickedNode, setClickedNode] = useState<Node >(graph.getNodes()[0])
+  // State to store the node that was clicked for the modal
+  const [clickedNode, setClickedNode] = useState<Node>(
+    getNodeById(graph, userId)
+  )
 
+  // State to store whether the modal is visible
   const [modalVisible, setModalVisible] = useState(false)
 
+  // Function to call when the modal is pressed
   const onModalPress = (uid: string) => {
-    const node = graph.getNodeById(uid)
+    const node = getNodeById(graph, uid)
```

Comment:
Here we have the same problem - commenting this type of code does not bring value, the variable/function names are already intuitive enough. One good example which clearly did need comments was the `ForceDirectedGraph` file, where not so-intuitive functions were defined. I am not sure whether the comments are needed here